### ISSUE TITLE
How to generate from lottie json with no background

### ISSUE THREAD

**Issue by jeb218** [transitive-bullshit/puppeteer-lottie-cli#16]
Even with -b "transparent" explicitly set, there is always a black background in the generated files. I know the background isn't embedded in the json, because I have also displayed the same files in lottie-react-native, and there is no background there.

I need to generate an mp4 from the lottie json, and then use ffmpeg to overlay the generated mp4 over another mp4 - but the lottie generated video has a large black background around it that covers the other video.

Do you have any tips for getting an mp4 out that has no background?

**Comment by transitive-bullshit** [comment]
mp4 videos don't support transparency.

Your best bet is to output a series of transparent png images and use that "image-*.png" type pattern as your overlay input.

**Comment by jeb218** [comment]
Yeah I realized the mistake in my thought process right after I posted. I tried to delete the issue but I guess they don't allow that. I was able to do the image series however

**Comment by gnipbao** [comment]
good issue! i have the same problem.

**Comment by khaledzag** [comment]
@transitive-bullshit u can use `-vcodec png` filter in ffmpeg command

**Comment by att1sb** [comment]
1. puppeteer-lottie -i ruledme.json -o 'frame-%d.png' --width 1024
2. ffmpeg -framerate 30 -i frame-%d.png -c:v gif -filter_complex "[0]split[s0][s1];[s0]palettegen[p];[s1][p]paletteuse" out.gif
3. ffmpeg -framerate 30 -i frame-%d.png -c:v libvpx-vp9 -b:v 0 -pix_fmt yuva420p out.webm

I went with webm instead of gif, easier.

**Comment by kenyerman** [comment]
# Problem with `mp4`

`mp4` is a container which supports multiple codecs (like: `h.264`, `h.265`). To have a transparent video the codec must support a pixel format which has `alpha` data (like: `yuva420p`, `rgba`).

Currently there are no `mp4` encoders which suppport pixel formats with `alpha` data.

# Workflow for creating transparent `webm`:

`VP9` codec supports `yuva420p`. This solution has an okayish browser support (if intrested).

## 1. create image squence:
```bash
npx puppeteer-lottie-cli -i animation.json -o 'animation/frame-%03d.png'
```

## 2. concatenate them with ffmpeg

ffmpeg can figure out from transparent png's which pixel format is needed.

```bash
ffmpeg -i animation/frame-%03d.png -b:v 800k animation.webm
```

# Other formats

`Apple ProRes 4444` format is the `prores_ks` codec with the `yuva444p10le` pixel format. The available containers are `mov`, `mkv`, `mxf` This supports transparency. However the filesize and support is far from optimal.

## 1. create image squence:
```bash
npx puppeteer-lottie-cli -i animation.json -o 'animation/frame-%03d.png'
```

## 2. concatenate them with ffmpeg
```
ffmpeg -i animation/frame-%03d.png -c:v prores_ks -pix_fmt yuva444p10le animation.mov
```
